webhook/resources/volume/validator.go

```diff
@@ -410,3 +431,54 @@ func (v *volumeValidator) canDisableRevisionCounter(engineImage string) (bool, e
 
 	return true, nil
 }
+
+func validateSnapshotMaxCount(snapshotMaxCount int) error {
+	if snapshotMaxCount < 2 || snapshotMaxCount > 250 {
+		return fmt.Errorf("snapshot max count should be between 2 to 250")
+	}
+	return nil
+}
+
+func validateSnapshotMaxSize(size, snapshotMaxSize int64) error {
+	if snapshotMaxSize != 0 && snapshotMaxSize < size*2 {
+		return fmt.Errorf("snapshot max size can be 0 or at least twice of volume size")
+	}
+	return nil
+}
+
+func (v *volumeValidator) validateUpdatingSnapshotMaxCountAndSize(newVolume *longhorn.Volume) error {
+	var (
+		currentSnapshotCount     int
+		currentTotalSnapshotSize int64
+		engine                   *longhorn.Engine
+	)
+	engines, err := v.ds.ListVolumeEngines(newVolume.Name)
+	if err != nil && !datastore.ErrorIsNotFound(err) {
+		return werror.NewInternalError(fmt.Sprintf("can't list engines for volume %s, err %v", newVolume.Name, err))
+	} else if len(engines) >= 2 {
+		return werror.NewInvalidError("can't update snapshotMaxCount or snapshotMaxSize during migration", "")
```

Comment:
Suggest checking `Status.CurrentMigrationNodeID` beforehand. However, ideally, we should allow users to set even it's during the migration, because it should eventually take effect. 

It's OK for this implementation now.